blivet/partitioning.py

```diff
@@ -1800,7 +1800,10 @@ def addRequest(self, req):
                              "LVRequest"))
 
         if req.device.metaDataSize:
-            self.pool -= int(self.vg.align(req.device.metaDataSize, roundup=True) / self.vg.peSize)
+            factor = 1
+            if hasattr(req.device, "lvs") and hasattr(req.device, "chunkSize"):  # thin pool
+                factor = 2
+            self.pool -= factor * int(self.vg.align(req.device.metaDataSize, roundup=True) / self.vg.peSize)
```

Comment:
Is metadata specification supported for some other device than a thin pool? This new ``if`` looks quite confusing to me. And once the thin pool grows and the extra space for metadata and pmspare is reserved, the VG reports ``freeSpace == 0``, right? Otherwise some LV grown later could potentially eat that reserved space for its own growth.

Reply:
Metadata size is an attribute of `LVMLogicalVolumeDevice` on rhel7-branch. I don't know what other types of LV might allow it, but I only know to double the space requirement for thin pools -- hence the complexity. Yes, the VG does report being completely full after growing the pool with this change.

Reply:
> Yes, the VG does report being completely full after growing the pool with this change.

Good.

> Metadata size is an attribute of LVMLogicalVolumeDevice on rhel7-branch.

Yes, but thin pools are instances of the ``LVMThinPoolDevice`` class so I think the check can be made simpler/more straightforward.

Reply:
So you think that even though the attr (`metaDataSize`) is in `LVMLogicalVolumeDevice` it is only used for thin pools? I wasn't sure if it was also used for something else like raid or cache lvs. (Or did I miss your point?)

Reply:
No, it's a very good idea to not limit the calculation to thin pools. I just think the check is weird -- why not use ``isinstance()``? And I would say it's better to always reserve the pmspare space whenever there's some metadata space required by any LV. The LVM RAID can use it for reshaping/restriping/...